### DISCUSSION TITLE
Does disabling unneeded modules improve performance?

### DISCUSSION THREAD

**Discussion by PatrickF1** [starship/starship#1920] [General]
Hi sorry in advance if this question has already been asked. Does disabling modules that would never get activated improve performance? For example, I don't use Conda but I wonder if the Starship code, because the Conda module is enabled by default, will check for $CONDA_DEFAULT_ENV, thereby slowing things down. Given that there's several modules I don't use, it might add up. Thanks!

**Comment by mickimnet** [comment]
For my use-cases, the performance of _Starship_ is no issue. I do only get lags if I'm in big repositories (like e. g. nixpkgs), and that is a known issue and will be handled in a future release.

**Comment by andytom** [comment]
Starship calls the modules in parallel so it is only as slow as the slowest module so unless there is a single module that is very slow disabling unused modules shouldn't have any noticeable effect. You can check how long each module took to generate using the `starship timings` command.
